### ISSUE TITLE
Replace uses of `time` (`precise_time_ns`) with `std::time`, `chrono`, or the new `time` crate

### ISSUE THREAD

**Issue by mrobinson** [servo/servo#30150]
We have a lot of uses of a very old version of the `time` crate. We should gradually replace all of these uses with calls to `std::time` or if that isn't sufficient with `chrono` or the updated version of the `time` crate. We already depend on those crates.

Some crates should not be modified such as:

 - `components/style`
 - `components/selectors`
 - `third_party/webrender`

These are vendored from Gecko and we need to make the fixes upstream.

**Comment by aadi58002** [comment]
I would like to work on this issue

**Comment by mrobinson** [comment]
> I would like to work on this issue

Okay. I recommend that you try to a small portion of the code to update and do it in pieces.

**Comment by aadi58002** [comment]
Hello there,
I would like know what your preferred way of doing this will be.
There are a lot of instances of the (Tm)[https://docs.rs/time/0.1.45/time/struct.Tm.html] struct which are used to compare time through out the program.It has both nanoseconds and seconds comparison which are taking place.I am using `Local::now().timestamp()` and `Local::now().timestamp_nanos()` from chrono to replace them.  I am not sure how we can clear show where the time used is in seconds or nanoseconds.

My suggestion is to move every time relates value from a i64 to a (Duration)[https://web.mit.edu/rust-lang_v1.25/arch/amd64_ubuntu1404/share/doc/rust/html/std/time/struct.Duration.html].

**Comment by aadi58002** [comment]
Hello there,
I would like to ask you about your preferred alternative to Tm struct from time.

It is frequently used for comparing time for various task ( like expiration of cookies ).
I currently replaced them with u64 which represent nanoseconds but in some place they use seconds for calculations. In struct definations it is not clear if a u64/i64 represents a seconds or nanosecond or millisecond. 

I am thinking of creating a Duration instance from std::time which we can get return from any range (seconds or nanoseconds or milliseconds) and they can be directly compared as well.

What do you think?

**Comment by jdm** [comment]
Duration seems like a reasonable choice if it works.

**Comment by mrobinson** [comment]
I agree. If we can use `Duration` that is much less ambiguous than using a number to hold seconds/milliseconds/nanoseconds.

**Comment by michaelgrigoryan25** [comment]
Is there any work around this? If not I can help.

**Comment by mrobinson** [comment]
> Is there any work around this? If not, I can help.

Any help would be greatly appreciated. Thanks!

**Comment by vijayksingh** [comment]
I want to work on this issue. I am thinking to start with one of the module inside components folder. Would be helpful if someone can let me know which one component would be good to start with.

**Comment by vijayksingh** [comment]
I want to work on this issue. I am thinking to start with one of the module inside components folder. Would be helpful if someone can let me know which one component would be good to start with.

**Comment by mrobinson** [comment]
Perhaps a good place to start would some of the DOM elements in the script crate such as `htmlformelement.rs`.

**Comment by reidswan** [comment]
Hey team 👋  to me this seems like the kind of issue multiple people could work on in tandem. Would we all be okay with me working on this as well?
 
If so, do we think `components/net/cookie.rs` or `components/devtools/actors/console.rs` could be good starting points?

**Comment by mrobinson** [comment]
I think it's fine to work on any part of this issue. A few people have volunteered, but so far no one has posted a WIP PR or an update or their progress.

**Comment by SlashScreen** [comment]
What more needs to be done for this?

**Comment by baimamboukar** [comment]
Hello @mrobinson, Is there something that's still to be done for this issue?

**Comment by mrobinson** [comment]
I no longer recommend this issue for folks looking to dive into Servo. Please look for other easy issues if that's your situation.

**Comment by atouchet** [comment]
@mrobinson is this finished now?

**Comment by mrobinson** [comment]
Yes, this is now complete.
